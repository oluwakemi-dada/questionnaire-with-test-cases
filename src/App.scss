@import url('https://fonts.googleapis.com/css2?family=Poppins:wght@300;400;500;600;700;800&display=swap');

html {
  font-size: 62.5%;
}

* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

body {
  font-family: 'Poppins', sans-serif;
  font-size: 1.6rem;
  line-height: 1.7;
  min-height: 100vh;
  overflow-x: hidden;
}

h1 {
  font-weight: 600;
  font-size: 2.8rem;

  @media (max-width: 700px) {
    font-size: 2.5rem;
  }

  @media (max-width: 450px) {
    font-size: 2.3rem;
  }
}

a {
  text-decoration: none;
  color: var(--dark-color);
}

input,
select,
textarea {
  color: #000;
  font-family: 'Poppins', sans-serif;
}

.text-center {
  text-align: center;
}
