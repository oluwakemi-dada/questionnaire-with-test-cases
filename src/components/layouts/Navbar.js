import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { BsMoonStarsFill } from 'react-icons/bs';
import { BsSunFill } from 'react-icons/bs';

const Wrapper = styled.nav`
  height: 8rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 5rem;
  position: relative;

  @media (max-width: 768px) {
    padding: 0 3rem;
  }

  @media (max-width: 450px) {
    padding: 0 2rem;
  }

  h1 {
    font-size: 2.5rem;
    color: #5e55ef;
  }
`;

export const ThemeToggle = styled.div`
  cursor: pointer;
  color: ${(props) => (props.theme === 'light' ? '#000' : '#fff')};
`;

const Navbar = ({ theme, themeToggler }) => {
  const currTheme = localStorage.getItem('theme');

  const onClickHandler = () => {
    themeToggler();
    if (currTheme === 'light') {
      localStorage.setItem('theme', 'dark');
    } else {
      localStorage.setItem('theme', 'light');
    }
  };
  return (
    <Wrapper>
      <h1>
        <Link to='/'>QA</Link>
      </h1>
      <ThemeToggle theme={theme} onClick={onClickHandler}>
        {theme === 'light' ? <BsSunFill /> : <BsMoonStarsFill />}
      </ThemeToggle>
    </Wrapper>
  );
};

export default Navbar;
