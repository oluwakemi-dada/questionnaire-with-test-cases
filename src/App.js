import React, { useState } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { lightTheme, darkTheme, GlobalStyles } from './themes';
import styled from 'styled-components';
import Navbar from './components/layouts/Navbar';
import HomeScreen from './screens/HomeScreen';
import SuccessScreen from './screens/SuccessScreen';
import NotFoundScreen from './screens/NotFoundScreen';
import './App.scss';

const StyledApp = styled.div`
  color: ${(props) => props.theme.fontColor};
`;

const App = () => {
  const currTheme = localStorage.getItem('theme');

  if (!currTheme) {
    localStorage.setItem('theme', 'light');
  }

  const [theme, setTheme] = useState(currTheme);

  const themeToggler = () => {
    theme === 'light' ? setTheme('dark') : setTheme('light');
  };

  return (
    <ThemeProvider theme={theme === 'light' ? lightTheme : darkTheme}>
      <GlobalStyles />
      <StyledApp>
        <Router>
          <Navbar theme={theme} themeToggler={themeToggler} />
          <Switch>
            <Route exact path='/' component={HomeScreen} />
            <Route path='/success' component={SuccessScreen} />
            <Route component={NotFoundScreen} />
          </Switch>
        </Router>
      </StyledApp>
    </ThemeProvider>
  );
};

export default App;
