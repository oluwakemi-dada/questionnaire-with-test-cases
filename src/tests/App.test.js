import React from 'react';
import { shallow } from 'enzyme';
import App from '../App';
import { mockLocalStorage } from './utils/mockLocalStorage';
import Navbar from '../components/layouts/Navbar';

const { getItemMock, setItemMock } = mockLocalStorage();
let wrapper;
beforeEach(() => {
  wrapper = shallow(<App />);
});

test('should render App correctly', () => {
  getItemMock.mockReturnValue('light');
  wrapper = shallow(<App />);
  expect(getItemMock).toHaveBeenLastCalledWith('theme');
  expect(wrapper).toMatchSnapshot();
});

test('should set theme to light in localstorage if no theme', () => {
  expect(setItemMock).toHaveBeenCalledWith('theme', 'light');
});

test('should update theme when themeToggler is called (light to dark)', () => {
  getItemMock.mockReturnValue('light');
  wrapper = shallow(<App />);
  expect(wrapper.find(Navbar).prop('theme')).toBe('light');
  wrapper.find(Navbar).prop('themeToggler')();
  expect(wrapper.find(Navbar).prop('theme')).toBe('dark');
});
