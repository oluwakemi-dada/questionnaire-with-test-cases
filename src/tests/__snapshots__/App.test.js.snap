// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly 1`] = `
<Component
  theme={
    Object {
      "body": "#fff",
      "fontColor": "#000",
    }
  }
>
  <Memo(l) />
  <styled.div>
    <BrowserRouter>
      <Navbar
        theme="light"
        themeToggler={[Function]}
      />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/success"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </BrowserRouter>
  </styled.div>
</Component>
`;
